src/Emulator/Peripherals/Peripherals/Network/K6xF_Ethernet.cs

```diff
@@ -0,0 +1,839 @@
+        {
+            var bytesArray = bytes.ToArray();
+
+            EnsureMinimumArrayLength(isCRCIncluded ? 64 : 60);
+
+            EthernetFrame frame;
+            var addCrc = !isCRCIncluded && !forwardCRCFromApplication.Value;
+            if (forwardCRCFromApplication.Value && !isCRCIncluded)
+            {
+                this.Log(LogLevel.Error, "CRC needs to be provided by the application but is missing");
+                return;
+            }
+
+            if(!Misc.TryCreateFrameOrLogWarning(this, bytesArray, out frame, addCrc))
+            {
+                return;
+            }
+
+            this.Log(LogLevel.Debug, "Sending packet, length {0}", frame.Bytes.Length);
+            FrameReady?.Invoke(frame);
+
+            void EnsureMinimumArrayLength(int length)
+            {
+                if(bytesArray.Length < length)
+                {
+                    Array.Resize(ref bytesArray, length);
+                }
+            }
+        }
+
+
+        private void SendFrames()
+        {
+            lock(innerLock)
+            {
+                var packetBytes = new List<byte>();
+                txDescriptorsQueue.CurrentDescriptor.Read();
+                while(txDescriptorsQueue.CurrentDescriptor.IsReady)
+                {
+                
+                    // fill packet with data from memory
+                    this.Log(LogLevel.Debug, "Buffer Length {0}", txDescriptorsQueue.CurrentDescriptor.Length);
+                    packetBytes.AddRange(txDescriptorsQueue.CurrentDescriptor.ReadBuffer());
+                    if(txDescriptorsQueue.CurrentDescriptor.IsLast)
+                    {
+                        // IncludeCRC is only valid for the buffer with IsLast set
+                        SendSingleFrame(packetBytes, !txDescriptorsQueue.CurrentDescriptor.IncludeCRC);
+                        packetBytes.Clear();
+                    }
+
+                    // Need to update the ready flag after processing
+                    txDescriptorsQueue.CurrentDescriptor.IsReady = false; // free it up
+                    txDescriptorsQueue.CurrentDescriptor.Update(); // write back to memory
+                    txDescriptorsQueue.GoToNextDescriptor();
+                }
+
+                this.Log(LogLevel.Debug, "Transmission completed");
+                isTransmissionStarted = false;
+
+                interruptManager.SetInterrupt(Interrupts.TransmitFrameInterrupt);
+                interruptManager.SetInterrupt(Interrupts.TransmitBufferInterrupt);
+            }
+        }
+
+
+        private readonly InterruptManager<Interrupts> interruptManager;
+        private DoubleWordRegisterCollection registers;
+        private readonly object innerLock;
+
+        // Fields needed for the internal logic
+        private bool isTransmissionStarted = false;
+
+        //EthernetControl
+        private IFlagRegisterField etherEnabled;
```

Comment:
Many of the fields defined here are not used at all. I would suggest changing them into tags.